src/scvi/nn/_base_components.py

```diff
@@ -173,14 +194,18 @@ def forward(self, x: torch.Tensor, *cat_list: int):
                             x = layer(x)
                     else:
                         if isinstance(layer, nn.Linear) and self.inject_into_layer(i):
-                            if x.dim() == 3:
-                                one_hot_cat_list_layer = [
-                                    o.unsqueeze(0).expand((x.size(0), o.size(0), o.size(1)))
-                                    for o in one_hot_cat_list
-                                ]
-                            else:
-                                one_hot_cat_list_layer = one_hot_cat_list
-                            x = torch.cat((x, *one_hot_cat_list_layer), dim=-1)
+                            if i > 0 and self.inject_covariates and cont_covs is not None:
+                                # Need to inject the continous covariates to hidden layers
+                                x = torch.cat((x, cont_covs), dim=-1)
+                            if self.batch_representation == "one-hot":
```

Comment:
this should be done irregardless of what is batch representation for all covariates that were passed as categorical